orderer/common/multichannel/registrar.go

```diff
@@ -1042,3 +1064,20 @@ func (r *Registrar) removeSystemChannel() error {
 
 	return nil
 }
+
+func (r *Registrar) removeLedgerAsync(channelID string) {
+	r.pendingRemoval[channelID] = true
+
+	go func() {
+		if err := r.ledgerFactory.Remove(channelID); err != nil {
+			r.lock.Lock()
+			delete(r.pendingRemoval, channelID)
+			r.lock.Unlock()
+			logger.Errorf("Ledger factory failed to remove empty ledger '%s', error: %s", channelID, err)
```

Comment:
I am thinking (see also comment below in the unit test), that in case of remove failure we need to keep the entry in the map, such that future join or remove to the same channel would fail? this would also alert the admin to the fact that he needs to intervene, but keeping the channel in a status "inactive".

Reply:
See updated logic